packages/axios-extension/src/abap/abap-service-provider.ts

```diff
@@ -213,4 +215,47 @@ export class AbapServiceProvider extends ServiceProvider {
 
         return this.services[subclassName] as T;
     }
+
+    /**
+     * Create a UI Service generator for the given business object.
+     *
+     * @param bo - business object
+     * @returns a UI Service generator
+     */
+    public async getUiServiceGenerator(bo: BusinessObject): Promise<UiServiceGenerator> {
+        const generatorService = await this.getAdtService<GeneratorService>(GeneratorService);
+        if (!generatorService) {
+            throw new Error('Generators are not support on this system');
+        }
+        const config = await generatorService.getUIServiceGeneratorConfig(bo.name);
+        const gen = this.createService<UiServiceGenerator>(this.getServiceUrlFromConfig(config), UiServiceGenerator);
+        gen.configure(config, bo);
+        return gen;
+    }
+
+    /**
+     * Get the service URL from the generator config.
+     *
+     * @param config - generator config
+     * @returns the service URL
+     */
+    private getServiceUrlFromConfig(config: GeneratorEntry): string {
+        // make code in this function defensive against undefined href
+        if (Array.isArray(config.link) && !config.link[0]?.href) {
+            throw new Error('No service URL found in the generator config');
+        }
+        const endIndex = config.link[0].href.indexOf(config.id) + config.id.length;
+        return config.link[0]?.href?.substring(0, endIndex);
```

Comment:
You already checked config.link[0].href is not undefined/null. There is no need to use '?' at this line.

Reply:
Thanks, done.